src/lib/support/utf8.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace Utf8 {
+
+/**
+ * Validate that the given span looks like a valid UTF-8 string:
+ *   - no embedded nulls (this is a Matter Spec requirement for encoding)
```

Comment:
Embedded nulls are allowed in Matter. It's **trailing** nulls that are not.

Reply:
Andrei - what's the spec ref for this? Cert tests just validate UTF-8, but if I can hook this to a spec requirement, we can test for it in cert too.